malfeasance/handler.go

```diff
@@ -31,37 +29,51 @@ var (
 	errInvalidProof  = fmt.Errorf("%w: invalid proof", pubsub.ErrValidationReject)
 )
 
+type (
+	MalfeasanceHandlerV1 func(ctx context.Context, data scale.Type) (types.NodeID, []string, error)
```

Comment:
It is already decoded by the `malfeasance` handler. I didn't change this, since it would require changing existing DTOs for malfeasance proofs from

```go
type Proof struct {
	// MultipleATXs | MultipleBallots | HareEquivocation | InvalidPostIndex
	Type uint8
	// AtxProof | BallotProof | HareProof | InvalidPostIndexProof
	Data scale.Type
}
```

to

```go
type Proof struct {
	Type uint8
	Data []byte
}
```

which I plan to do with V2 of malfeasance proofs.

The reason I use `scale.Type` and not `any` is because `any` allows passing anything, while `scale.Type` is more restrictive (i.e. only types with scale encoding can be passed). It's not ideal but I don't think I can be more more specific without breaking existing code 😕

Reply:
The only way to make this work is to have the `ProofData` interface (`Proof` is already a type) extend the `scale.Type` interface and put it in the `malfeasance/wire` package.